### PR TITLE
#187: Client page adding postcode and phone number filter

### PR DESCRIPTION
**Pull request by KylePUG**
## What's changed
- Added the "Phone Number" column in the "Clients" page
- Added filters for "Phone Number" and "Postcode" 

## Screenshots / Videos
| Before     | After      |
|------------|------------|
|image|image|

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database... (no changes)

### CODE REVIEW

File: src/app/clients/ClientsPage.tsx
Comment (AmeliaIsCoding) [inline src/app/clients/ClientsPage.tsx]: lets change this to be consistent with the parcels table: `["phoneNumber", "Phone"],`

File: src/app/clients/ClientsPage.tsx
Comment (AmeliaIsCoding) [inline src/app/clients/ClientsPage.tsx]: I recommend copying and pasting the postcode search code from the parcels table. It's not quite as simple as this, because we need to be able to search for a `null` postcode in the backend when "NFA" is input into the filter, so it's not a direct mapping between the state and the value we put into the the `.ilike`.

File: src/app/clients/ClientsPage.tsx
Comment (AmeliaIsCoding) [inline src/app/clients/ClientsPage.tsx]: change this to `"Phone"` please (consistency with parcels page)

**Review by AmeliaIsCoding — CHANGES_REQUESTED** [review]
Looks good! Some changes needed

**Comment by KylePUG** [comment]
Changes:

 - Changed the naming conventions from "Phone Number" to "Phone"
 - Included the cases to find `null` postcodes using "NFA" in the filter
 - Included lowercase search

QA has been redocumented to include these changes as well

**Review by AmeliaIsCoding — COMMENTED** [review]
LGTM, feel free to hand over to Ricky/Kristina for their review

File: src/app/clients/ClientsPage.tsx
Comment (8kristy) [inline src/app/clients/ClientsPage.tsx]: It looks like it's exactly the same as the parcel one except the database field name (?), in which case I think it might be good to pull it out into a separate helper function somewhere and pass the field name as a parameter (the `common` folder looks like a good place for it maybe?)

**Review by 8kristy — CHANGES_REQUESTED** [review]
Just one little change

File: src/app/clients/ClientsPage.tsx
Comment (KylePUG) [reply src/app/clients/ClientsPage.tsx]: This makes a lot of sense - I was trying this today and had to deal with some merge conflicts from some refactoring in the dev branch. I think this might be more complicated now that the functions in `filters.ts` under clientsTable and parcelsTable have different types. I need to look into types and generics - might take me some time but I'm happy to do it! Let me know if you want me to pick this up on Monday! :)

Comment (KylePUG) [reply src/app/clients/ClientsPage.tsx]: Ignore the above message - I have just done this. Ready for review!

File: src/common/databaseFilters.ts
Comment (AmeliaIsCoding) [inline src/common/databaseFilters.ts]: why do we need String here?

File: src/common/databaseFilters.ts
Comment (AmeliaIsCoding) [inline src/common/databaseFilters.ts]: I feel like, we could call this fullNameSearch, and similarly for the others

File: src/app/parcels/parcelsTable/filters.ts
Comment (AmeliaIsCoding) [inline src/app/parcels/parcelsTable/filters.ts]: I feel like if we change the `phoneNumberMethodHelper` to be `phoneNumberSearch`, we could rename this to `parcelsPhoneNumberSearch` which I think is a bit clearer. Similarly for the others

File: src/common/databaseFilters.ts
Comment (AmeliaIsCoding) [inline src/common/databaseFilters.ts:1]: are there any other filters search methods that are common between tables? the users table also has filters and is server side paginated

File: src/common/databaseFilters.ts
Comment (AmeliaIsCoding) [inline src/common/databaseFilters.ts]: It would be really good if you could check that this conforms to [VFB-211](https://softwiretech.atlassian.net/jira/software/c/projects/VFB/boards/862?selectedIssue=VFB-211)

i.e., if we set columnLabel to be a string that **is not** a DB column for the specific table, what happens?

**Review by AmeliaIsCoding — CHANGES_REQUESTED** [review]
well done on generalising the filter search methods! a few small comments

File: src/common/databaseFilters.ts
Comment (KylePUG) [reply src/common/databaseFilters.ts]: thanks for noting this! an artifact from a previous version of the code - it is unnecessary

File: src/app/parcels/parcelsTable/filters.ts
Comment (AmeliaIsCoding) [inline src/app/parcels/parcelsTable/filters.ts:24]: let's be consistent between parcels vs parcel - i don't really have a preference but just stick with one :)

**Review by AmeliaIsCoding — CHANGES_REQUESTED** [review]
1 tiny thing then request a review from kristina pls :)

**Review by 8kristy — APPROVED** [review]
Happy to get this merged after making the names consistent as Amelia suggested

**Comment by KylePUG** [comment]
Merging is blocked on this PR for some reason. @8kristy do you know why?

**Comment by 8kristy** [comment]
Think it might be because it's waiting for changes and approval from @AmeliaIsCoding ? Could you approve this please I can't find a setting that allows to merge if owner approved but someone else requires changes

**Review by AmeliaIsCoding — APPROVED** [review]

**Pull request merged**
